packages/playwright-core/src/server/webkit/wkProvisionalPage.ts

```diff
@@ -62,6 +78,29 @@ export class WKProvisionalPage {
     this._wkPage._onFrameAttached(this._mainFrameId, null);
   }
 
+  private _onRequestWillBeSent(event: Protocol.Network.requestWillBeSentPayload) {
+    if (this._coopNavigationRequest && this._coopNavigationRequest.url() === event.request.url) {
```

Comment:
IIUC, the first request must be the `coopNavigationRequest`? Perhaps we should assert that urls are equal instead of comparing?

Reply:
That is correct. I'm not sure about turning this into an assert, in case something changes and the page starts cancelling the cross process provisional load a bit later or something like that it could start firing.